docs/src/content/docs/cookbook/dexes/dedust.mdx

```diff
@@ -7,8 +7,434 @@ sidebar:
+    forwardPayload: Cell?; // slightly adjusted
+}
+```
+
+:::note[Useful links:]
+
+  [Retrieving Jetton wallet address in TON Docs](https://docs.ton.org/develop/dapps/asset-processing/jettons#retrieving-jetton-wallet-addresses-for-a-given-user)\
+  [How to calculate user's Jetton wallet address (offline)?](https://docs.ton.org/v3/guidelines/dapps/cookbook#how-to-calculate-users-jetton-wallet-address-offline)
+
+:::
+
+## Liquidity Provisioning
+
+To provide liquidity to a particular DeDust pool, you must provide both assets. The pool will then issue special _LP tokens_ to the depositor's address.
+
+Read more about liquidity provisioning in the [DeDust documentation](https://docs.dedust.io/docs/liquidity-provisioning).
+
+```tact
+import "@stdlib/deploy";
+
+/// https://docs.dedust.io/reference/tlb-schemes#message-deposit_liquidity-1
+message(0x40e108d6) JettonDepositLiquidity {
+    // Pool type: 0 for volatile, 1 for stable
+    // Volatile pool is based on the "Constant Product" formula
+    // Stable-swap pool is optimized for assets of near-equal value,
+    // e.g. USDT/USDC, TON/stTON, etc.
+    poolType: Int as uint1;
+
+    // Provided assets
+    asset0: Asset;
+    asset1: Asset;
+
+    // Minimal amount of LP tokens to be received
+    // If there's less liquidity provided, the provisioning will be rejected
+    // Defaults to 0, makes this value ignored
+    minimalLpAmount: Int as coins = 0;
+
+    // Target amount of the first asset
+    targetBalances0: Int as coins;
+
+    // Target amount of the second asset
+    targetBalances1: Int as coins;
+
+    // Custom payload to attack to the transaction if the provisioning is successful
+    // Defaults to `null`, which means no payload
+    fulfillPayload: Cell? = null;
+
+    // Custom payload to attack to the transaction if the provisioning is rejected
+    // Defaults to `null`, which means no payload
+    rejectPayload: Cell? = null;
+}
+
+/// https://docs.dedust.io/reference/tlb-schemes#message-deposit_liquidity
+message(0xd55e4686) NativeDepositLiquidity {
+    // Unique identifier used to trace transactions across multiple contracts
+    // Defaults to 0, which means we don't mark messages to trace their chains
+    queryId: Int as uint64 = 0;
+
+    // Toncoin amount for the deposit
+    amount: Int as coins;
+
+    // Inlined fields of JettonDepositLiquidity Message without the opcode prefix
+    poolType: Int as uint1;
+    asset0: Asset;
+    asset1: Asset;
+    minimalLpAmount: Int as coins = 0;
+    targetBalances0: Int as coins;
+    targetBalances1: Int as coins;
+    fulfillPayload: Cell? = null;
+    rejectPayload: Cell? = null;
+}
+
+/// https://docs.dedust.io/reference/tlb-schemes#asset
+struct Asset {
+    // Specify 0 for native (TON) and omit all following fields
+    // Specify 1 for Jetton and then you must set non-null values for the following fields
+    type: Int as uint4;
+
+    workchain: Int? as uint8 = null;
+    address: Int? as uint256 = null;
+}
```

Comment:
According to dedust docs, workchain and address are exact values, not Maybe uint8 and Maybe uint256. Moreover, it is possible to serialize this structure automatically. 
1st way (kind of Hack)
```
struct Asset {
    // Specify 0 for native (TON) and omit all following fields
    // Specify 1 for Jetton and then you must set non-null values for the following fields
    type: Int as uint3 = 0; //Not uint4, should always be zero
    info: JettonInfo?;
}
struct JettonInfo {
    workchain: Int as uint8 = null;
    address: Int as uint256 = null;
}
```
Or
```
struct Asset {
    type: Int as uint4 = 0;
    workchain: Int as uint8 = 0;
    address: Int as uint256 = 0;
}
```
In second case you can just leave 0 in workchain and address I am quite sure this won't affect any processing. But this is definitely better than making this fields optional and then serializing them manually.

Reply:
Yeah, let's stay with the manual approach until we'll be able to re-write this somewhat different in the future, as @Gusarich [noted here](https://github.com/tact-lang/tact/pull/954#pullrequestreview-2440314824)